test/integration/webhook/v1alpha2/resourceflavor_test.go

```diff
@@ -76,6 +83,43 @@ var _ = ginkgo.Describe("ResourceFlavor Webhook", func() {
 		})
 	})
 
+	ginkgo.DescribeTable("invalid number of properties", func(taintsCount int, nodeSelectorCount int, isInvalid bool) {
+		resourceFlavor := func() *kueue.ResourceFlavor {
+			rf := testing.MakeResourceFlavor("resource-flavor")
+			for i := 0; i < taintsCount; i++ {
+				rf.Taint(corev1.Taint{
+					Key:    fmt.Sprintf("t%d", i),
+					Effect: corev1.TaintEffectNoExecute,
+				})
+			}
+
+			m := make(map[string]string)
+			for i := 0; i < nodeSelectorCount; i++ {
+				m[fmt.Sprintf("l%d", i)] = ""
+			}
+
+			return rf.MultiLabels(m).Obj()
+		}()
+		err := k8sClient.Create(ctx, resourceFlavor)
+		if isInvalid {
+			gomega.Expect(err).To(gomega.HaveOccurred())
+			gomega.Expect(errors.IsInvalid(err)).To(gomega.BeTrue(), "error: %v", err)
+		} else {
+			gomega.Expect(err).To(gomega.Succeed())
+			defer func() {
+				var rf kueue.ResourceFlavor
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)).To(gomega.Succeed())
+				controllerutil.RemoveFinalizer(&rf, kueue.ResourceInUseFinalizerName)
+				gomega.Expect(k8sClient.Update(ctx, &rf)).Should(gomega.Succeed())
+				framework.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, resourceFlavor, true)
```

Comment:
I think this should be enough. The controller should remove the finalizer, given that the RF is not in use.

```suggestion
				framework.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, resourceFlavor, true)
```

Reply:
That's because the controller is not running in this test.
Enable it please: `core.SetupControllers`

Reply:
If we enable `core.SetupControllers` in `test/integration/webhook/v1alpha2/suite_test.go`, we got many update conflict event in `clusterqueue_test.go` and `workload_test.go`, like
```
[FAILED] error: Operation cannot be fulfilled on clusterqueues.kueue.x-k8s.io "cluster-queue": the object has been modified; please apply your changes to the latest version and try again
Expected
      <bool>: false
  to be true
  In [It] at: /go/src/kueue/test/integration/webhook/v1alpha2/clusterqueue_test.go:120
```

```
[FAILED] error: Operation cannot be fulfilled on workloads.kueue.x-k8s.io "workload-test": the object has been modified; please apply your changes to the latest version and try again
  Expected
      <bool>: false
  to be true
  In [It] at: /go/src/kueue/test/integration/webhook/v1alpha2/workload_test.go:169
```

Reply:
No idea what those failing lines are trying to do, but, once you enable the controllers:

- you shouldn't be trying to delete the finalizers in the test code.
- any updates to objects should have retries. We do this for other integration tests.

Reply:
I had tried to do so but failed as mentioned above. I will try to finish this in a follow up. Thank you for your patience and guidance!

Reply:
example of retrying https://github.com/kubernetes-sigs/kueue/blob/7b8316fa811f4afc75e22803e33b3061a030afb8/test/integration/controller/core/localqueue_controller_test.go#L89-L95